### PR TITLE
Move operation to library

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Move operations from client to library so that everyone can use them.
Long term goal: 
- have tests for all operations on library, against real server
- have tests on client against mocked library

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

File: src/com/owncloud/android/lib/resources/comments/CommentFileRemoteOperation.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
+/* ownCloud Android Library is available under MIT license
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/lib/resources/comments/CommentFileRemoteOperation.java]: shouldn't we rename it from oC to Nc?

File: src/com/owncloud/android/lib/resources/files/RestoreFileVersionRemoteOperation.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/*
+ * Nextcloud Android client application
+ *
+ * @author Tobias Kaminsky
+ * Copyright (C) 2018 Tobias Kaminsky
+ * Copyright (C) 2018 Nextcloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/lib/resources/files/RestoreFileVersionRemoteOperation.java]: The lib project is licensed under the MIT License!

File: src/com/owncloud/android/lib/resources/trashbin/EmptyTrashbinRemoteOperation.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
+/*
+ * Nextcloud Android client application
+ *
+ * @author Tobias Kaminsky
+ * Copyright (C) 2018 Tobias Kaminsky
+ * Copyright (C) 2018 Nextcloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/lib/resources/trashbin/EmptyTrashbinRemoteOperation.java]: same here should be MIT license

File: src/com/owncloud/android/lib/resources/trashbin/RestoreTrashbinFileRemoteOperation.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
+/*
+ * Nextcloud Android client application
+ *
+ * @author Tobias Kaminsky
+ * Copyright (C) 2018 Tobias Kaminsky
+ * Copyright (C) 2018 Nextcloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/lib/resources/trashbin/RestoreTrashbinFileRemoteOperation.java]: MIT license

**Comment by AndyScherzinger** [comment]
👍 @tobiasKaminsky just some minor comments while the license comments need to be fixed (in my opinion) :) Other than that: all good

File: src/com/owncloud/android/lib/resources/comments/CommentFileRemoteOperation.java (hunk 1 above)
Comment (tobiasKaminsky) [reply src/com/owncloud/android/lib/resources/comments/CommentFileRemoteOperation.java]: Correct, for new added files we can do this.

File: src/com/owncloud/android/lib/resources/files/DownloadFileRemoteOperation.java [hunk 5]
```diff
@@ -97,8 +96,8 @@ protected RemoteOperationResult run(OwnCloudClient client) {
     }
 
 
-    protected int downloadFile(OwnCloudClient client, File targetFile) throws HttpException,
-        IOException, OperationCancelledException {
+    protected int downloadFile(OwnCloudClient client, File targetFile) throws
```
[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply src/com/owncloud/android/lib/resources/files/DownloadFileRemoteOperation.java]: Nothing we can do in this PR…

**Pull request merged**